lib/Frontend/CompilerInvocation.cpp

```diff
@@ -118,94 +118,80 @@ class ArgsToFrontendInputsConverter {
   FrontendInputs &Inputs;
 
   Arg const *const FilelistPathArg;
+  Arg const *const PrimaryFilelistPathArg;
 
-  std::unique_ptr<llvm::MemoryBuffer> FilelistBuffer;
-
-  StringRef filelistPath() { return FilelistPathArg->getValue(); }
+  SmallVector<std::unique_ptr<llvm::MemoryBuffer>, 4> BuffersToKeepAlive;
 
   llvm::SetVector<StringRef> Files;
+  std::set<StringRef> PrimaryFiles;
```

Comment:
llvm::SmallSet?

Reply:
Done.

Reply:
Whoops! SmallSet would not compile with a size of 200, and we're expected 150 of these easy. Am sticking with a std::set.

Reply:
Ah, I was thinking about very small projects where there's just one or two primaries per core, but I guess it doesn't matter too much!